packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1019,9 +1036,16 @@ export class GarbageCollector implements IGarbageCollector {
 
         // If we are running in GC test mode, delete objects for unused routes. This enables testing scenarios
         // involving access to deleted data.
-        if (this.testMode) {
-            this.runtime.updateUnusedRoutes(gcResult.deletedNodeIds);
+        if (this.sweepDataStoresMode) {
+            const allRoutes = gcResult.deletedNodeIds.concat(gcResult.referencedNodeIds);
+            this.validateSweepRoutes(sweepReadyNodes, allRoutes);
+            const removedRoutes = this.runtime.updateUnusedRoutes(sweepReadyNodes);
```

Comment:
Deleting nodes should be incremental, i.e., delete should only be called for nodes that haven't already been deleted. Currently, this will call delete for all nodes that are sweep ready, including those that have already been deleted.

Reply:
I thought the sweepReadyNodes would not include nodes already deleted as those nodes should have been removed previously from the `updateSweptNodeState` (which maybe needs a better name like `deleteSweptNodes`)

Reply:
You may be right about that. Let me give it another look.

Reply:
Maybe this is another point for validation, I'll put a comment here and we can decide whether or not we actually validate in a separate PR

Reply:
A lot of this code is written based on assumptions that the runtime works as expected, and it is honestly a little scary unless you know how the whole system works. Having guide rail validations makes these functions a little safer to call

Reply:
Yeah, you are right. Having validations will make this safer. Let's work on that once we write the core sweep logic.

Reply:
https://dev.azure.com/fluidframework/internal/_workitems/edit/3248